File: springwebfluxtutorial/src/main/java/com/qiusuo/webfluxtutorial/config/R2DBCConfig.java
package com.qiusuo.webfluxtutorial.config;

import io.r2dbc.spi.ConnectionFactories;
import io.r2dbc.spi.ConnectionFactory;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.r2dbc.config.AbstractR2dbcConfiguration;

@Configuration
public class R2DBCConfig extends AbstractR2dbcConfiguration {
    public ConnectionFactory connectionFactory() {
        ConnectionFactory connectionFactory = ConnectionFactories
                .get("r2dbc:postgresql://localhost:5432/qiusuo");
        return connectionFactory;
    }
    /*
    public ConnectionFactory connectionFactory() {
        return new PostgresqlConnectionFactory(
                PostgresqlConnectionConfiguration.builder()
                        .host("localhost")
                        .port(5432)
                        .username("qiusuo")
                        .database("qiusuo")
                        .password("qiusuo")
                        .build());
    }
    */
}

File: springwebfluxtutorial/src/main/kotlin/com/qiusuo/webfluxtutorial/config/R2DBCConfig.kt
package com.qiusuo.webfluxtutorial.config

import io.r2dbc.postgresql.PostgresqlConnectionConfiguration
import io.r2dbc.postgresql.PostgresqlConnectionFactory
import io.r2dbc.spi.ConnectionFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.data.r2dbc.config.AbstractR2dbcConfiguration
import org.springframework.data.r2dbc.repository.config.EnableR2dbcRepositories

@Configuration
@EnableR2dbcRepositories
class R2DBCConfig : AbstractR2dbcConfiguration() {
    @Bean
    override fun connectionFactory(): ConnectionFactory {
        return PostgresqlConnectionFactory(
            PostgresqlConnectionConfiguration.builder()
                .host("localhost")
                .port(5432)
                .username("qiusuo")
                .database("qiusuo")
                .password("qiusuo")
                .build()
        )
    }
}